### ISSUE TITLE
Not compatible with the latest OpenWRT/luci release 18.06.0

### ISSUE THREAD

**Issue by caesarw** [apollo-ng/luci-theme-darkmatter#19]
The theme is not compatible with the latest version of OpenWRT and luci (18.06.0), which caused the Overview panel to be a mess (see screenshots below). 

![screenshots](https://user-images.githubusercontent.com/11749340/43688343-5e565ba8-9919-11e8-8458-c6051ce4b530.PNG)

Any fixes on this? 
Thanks!

**Comment by xinfli** [comment]
Seems the most import is that the layout of Luci has been changed from table layout to div/span layout, then most of the css styles could not be found in darkmatter css files, so it caused mess view in most web page; I think it's not easy to fix.

Another problem is that a new .js need be loaded in Luci, it's missing in darkmatter template file (You can see many errors in web console); it can be fixed by modify /usr/lib/lua/luci/view/themes/darkmatter/header.htm, just change line
`    <script src="<%=resource%>/xhr.js?v=git-17.010.73729-fd88e48"></script>`
to 
`    <script src="<%=resource%>/cbi.js?v=git-18.210.69179-6df9a57"></script>
    <script src="<%=resource%>/xhr.js?v=git-18.210.69179-6df9a57"></script>`

**Comment by genX75** [comment]
Experiencing the same issue with this theme. A quick look around and it appears a lot of classes have changed.

**Comment by chron0** [comment]
ye, lot of things have changed and I'm still bound to https://github.com/domino-team/lede-ar300m, since there is still no support for the ar-300m. I know that darkmatter is popular and I'm rather unhappy that it's not working for you guys anymore but I don't have the time time to rework everything again for something that I can't even use on my hardware :/

**Comment by genX75** [comment]
Hey! thanks for the heads up. 

I'm considering doing the bulk of the work if I can get a hold of some documentation to help me. If you can point me in any sort of direction to make things easier I'd appreciate it.

**Comment by chron0** [comment]
really not much to it:

https://github.com/apollo-ng/luci-theme-darkmatter/blob/master/src/style.css

"just" need to match up the classes again, sometimes there haven't been any in the html so you need to count elements or use other CSS/JS trickery to get a match, it's a big PITA and time consuming. I just used the browser DOM/CSS and did one page at a time in the browser to see how it looked and the put the changed CSS/JS into the files in src.

**Comment by genX75** [comment]
I was looking at it and started doing it the brute force way. But I figured there must of been a better way. Looks like I was wrong sadly. I'll see what I can do over time.

**Comment by genX75** [comment]
Best thing I could dig up was this so far: https://github.com/openwrt/luci/wiki/Documentation

**Comment by chron0** [comment]
hehe - ye, brute force it was. OpenWRTs 4MB flash size constraint weighs in heavily on freedom to theme. I've tried to bring it down as much as I could, but jquery takes its toll in size. The basic html coming from luci is not very theming friendly anyways, due to size constraints, dev legacy and only very few common classes/groups to keep a style consistent. When you factor in other opkgs which bring their own luci extension it gets even worse :) It would be great to get rid of jquery as well.

**Comment by cantenna** [comment]
I wish lede/openwrt folks would adopt this theme, it's that good! I'm reaming on an older build because of this.

**Comment by cantenna** [comment]
ヽ(´ー｀)ノ

**Comment by boktai1000** [comment]
Are there any unofficial IPKs updates of this theme anywhere? I checked every fork available but they're all outdated.

In the meantime I've taken a liking to the theme "Rosy", but Darkmatter would be nice. Right now it seems like the theme is abandonware though/will not update for latest release. 

In regards to a theme being officially adopted in the mainline OpenWrt release- the LEDE/OpenWrt folks won't adopt a theme unless the theme creator is willing to maintain it if there are any issues in the future, as was the catch for Rosy after reading some GitHub comments before they accepted it as they don't want to introduce a theme and then take on responsibility of updating it if it breaks or has issues with any devices/browsers/etc.

**Comment by birkhofflee** [comment]
+1 Please update

**Comment by john-shine** [comment]
@boktai1000 
A quick fix: https://github.com/john-shine/luci-theme-darkmatter, hope it help

**Comment by chron0** [comment]
wanna send a PR?

**Comment by chron0** [comment]
> In regards to a theme being officially adopted in the mainline OpenWrt release- the LEDE/OpenWrt folks won't adopt a theme unless the theme creator is willing to maintain it if there are any issues in the future, as was the catch for Rosy after reading some GitHub comments before they accepted it as they don't want to introduce a theme and then take on responsibility of updating it if it breaks or has issues with any devices/browsers/etc.

We can't really hope for that to happen since the openwrt crowd seems to have changed over the last couple of years and started to make it very hard for people to get anything in there and also offers no dialogue on how we could make UX things less archaic while still keeping the 4MB constraint for a lot of devices out there. I doubt there are many people who will promise "maintenance over many years to come" - which certainly is quite a commitment. Begging for inclusion AND having to promise maintenance WHILE having to deal with constant changes coming from upstream potentially breaking UX again and again is nothing I am willing to do.

I'm just happy someone else picked it up and fixed it for now since there seem to be a lot of darkmatter installations out there and all would be happy to have it working with recent releases. Let's just keep it going like this, independent of the openwrt bubble.

**Comment by john-shine** [comment]
@chron0 PR send and extra work is necessary.

**Comment by chron0** [comment]
@john-shine <3 - thank you, merged.

**Comment by chron0** [comment]
@CaesarW could you test if it works again pls?

**Comment by caesarw** [comment]
@chron0 yeah I'll do a test later. Thanks all your guys!

**Comment by boktai1000** [comment]
Curious question I have- does this patch make it so the theme is compatible with versions both before and after 18.06, or is the theme now only compatible with 18.06 and beyond? Thanks!

**Comment by chron0** [comment]
I wouldn't count on it, but I can't test since there is still (3y later) no nand openwrt for my gl.inets. If you have to deploy for <18 just use the commit before.

**Comment by boktai1000** [comment]
@chron0 based on earlier comments I thought you were stuck on an older version and was mostly curious if these changes were backwards compatible with whatever you were running, as well as forward compatible with 18.06 and on.

**Comment by chron0** [comment]
yeah, I'm stuck on old but since I can't use my hardware with that either (support for hardware interrupts has also been removed which I need for PPS) I'm currently not building new images for them. Thats technically the main factor why I haven't touched darkmatter for openwrt in a while. So I can't simply test and confirm either way :/

**Comment by JeanGionet** [comment]
has anymore "fixed" this theme yet? I know I could update the styles file to update everything.. however that seems like a VERY time consuming thing to do! ;) Thanks

**Comment by chron0** [comment]
yes, VERY - hence my personal reluctance to do this again every time - while the upstream crowd is bitching about bullshit. iirc @john-shine did an update not long ago, see above...

**Comment by JeanGionet** [comment]
> yes, VERY - hence my personal reluctance to do this again every time - while the upstream crowd is bitching about bullshit. iirc @john-shine did an update not long ago, see above...

thanks for the quick reply! It does seem like the latest release from @john-shine works!
I snatched it from https://github.com/john-shine/luci-theme-darkmatter/releases/tag/v0.3-beta.2

thanks!

**Comment by chron0** [comment]
awesome, thanks for testing <3

**Comment by chron0** [comment]
@john-shine: is that one newer than the one in the last PR? If so, please send another PR so we can keep it up to date here as well.

**Comment by john-shine** [comment]
small amendment forget to PR, I had just send it.

**Comment by mrlemonde** [comment]
thanks for this nice work .. the last version has a bug where:
1 - there are no data available in overview page for Memory bears.
2 - the system pages only ( general settings)page has all the information of another pages as well . and in the up i have menu about another pages that it is unchangble menue.
same as for realtime Graph.
3 - DHCP and DNS , only general setting page has information.
